### ISSUE TITLE
output category 

### ISSUE THREAD

**Issue by betterze** [QuocThangNguyen/deep-metric-learning-tsinghua-dogs#2]
Dear QuocThangNguyen,

Thank you for your great work about Tsinghua Dogs dataset, I really like it.

In the main pape, you provide pretrained model, this model only provide the image embedding for retrieval rather than output the dog category. Is there a way to get the dog category given an image? 

Thank you for your help.

Best Wishes,

Alex

**Comment by QuocThangNguyen** [comment]
Hi @betterze ,

Thanks for your compliment. You are right that the pre-trained models only return the image embedding. To infer the category of an input image, I compare the input embedding with the reference embeddings using KNN. You can get the output dog category by following the instruction [here](https://github.com/QuocThangNguyen/deep-metric-learning-tsinghua-dogs#4-inference).

Also take a look at those [lines](https://github.com/QuocThangNguyen/deep-metric-learning-tsinghua-dogs/blob/fca6beb556d8c47d9e49e296262729c717c77bb7/src/infer.py#L276-L283), the `retrieved_labels` is a list of categories given an input image.

Hope this help
